### DISCUSSION TITLE
How to convert any csv to an Xarray with chosen dimension ?

### DISCUSSION THREAD

**Discussion by predicteur** [pydata/xarray#6631] [Ideas]
Analysis of the structure of a csv file makes it possible to transform it automatically or manually into a matrix of the chosen dimension while retaining all the information (see attachment and link below) . 

This property is very interesting to me because it makes it easy to generate Xarray. Conversely, it allows to generate csv files which preserve the Xarray structure.

A first version of this module is now available, and I would like to have your feedback (and also use cases) to validate this solution which contributes to making Xarray _**“an even more flexible and powerful tool for multidimensional data analysis”**_.

[csv_to_xarray.pdf](https://github.com/pydata/xarray/files/8755338/csv_to_xarray.pdf)

[https://loco-philippe.github.io/ES/ilist.html](https://loco-philippe.github.io/ES/ilist.html)

**Comment by RichardScottOZ** [comment]
Is there a repository?
